classes/Cart.php

```diff
@@ -919,10 +919,6 @@ protected function applyProductCalculations($row, $shopContext, $productQuantity
             $shopContext->shop = new Shop((int) $row['id_shop']);
         }
 
-        $address = Address::initialize($address_id, true);
```

Comment:
Are you sure this does not apply global changes in config and might affect calculations?

Reply:
This method stores its return value in cache, but it's only used within it. 

The only risky side effect I see is that when `$address_id` is empty, it will use contextual information to find out the address, so theoretically, if you call `Address::initialize(0)` here, then you change the context, then the call this method again, you will get the results according to the state of the context at the time of the first method call, it will not reflect the changes to context (because of cache). 

That said, I don't think the risk of these variables being changed later in the execution flow is too high:
- `customer->geoloc_id_country`
- `customer->id_state`
- `customer->postcode`
- `country->id`.